File: 1. Two Sum/TwoSum.py
#Brute Force
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        for i in range (len(nums)):
            num1= nums[i]
            for j in range (i+1, len(nums)):
                num2 = nums[j]
                if num1 + num2 == target:
                    return [i, j]
 

 #O(n) Solution
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        seen = {}
        for i,num in enumerate(nums):
            complement = target - num
            if complement in seen:
                return [seen[complement], i]
            seen[num]=i

File: 1. Two Sum/TwoSum.js
// Brute Force
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    for (let i=0; i<nums.length; i++){
        let num1=nums[i];
        for(let j=i+1; j<nums.length; j++){
            let num2=nums[j];
            if (num1+num2==target){
                return ([i,j]);
            }
        }
    }
};


 //O(n) Solution
 /**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const seen = new Map();
    for (let i=0; i<nums.length; i++ ){
        complement = target - nums[i];
        if (seen.has(complement)){
            return [seen.get(complement), i];
        }
        seen.set(nums[i])=i;
    }

};